pkg/platform/abstract/project/external/leader/iguazio/client.go

```diff
@@ -23,13 +33,140 @@ func NewClient(parentLogger logger.Logger, platformConfiguration *platformconfig
 }
 
 func (c *Client) Create(createProjectOptions *platform.CreateProjectOptions) error {
-	return nuclio.ErrNotImplemented
+	c.logger.DebugWith("Sending create project request to leader",
+		"name", createProjectOptions.ProjectConfig.Meta.Name,
+		"namespace", createProjectOptions.ProjectConfig.Meta.Namespace)
+
+	// generate request body
+	body, err := c.generateProjectRequestBody(createProjectOptions.ProjectConfig)
+	if err != nil {
+		return errors.Wrap(err, "Failed to generate project request body")
+	}
+
+	// send the request
+	headers := c.generateCommonRequestHeaders()
+	responseBody, _, err := common.SendHTTPRequest(http.MethodPost,
+		fmt.Sprintf("%s/%s", c.platformConfiguration.ProjectsLeader.APIAddress, "projects"),
+		body,
+		headers,
+		[]*http.Cookie{createProjectOptions.SessionCookie},
+		http.StatusCreated,
+		true)
+	if err != nil {
+		return errors.Wrap(err, "Failed to send request to leader")
+	}
+
+	c.logger.DebugWith("Successfully sent create project request to leader",
+		"name", createProjectOptions.ProjectConfig.Meta.Name,
+		"namespace", createProjectOptions.ProjectConfig.Meta.Namespace,
+		"responseBody", string(responseBody))
+
+	return nil
 }
 
 func (c *Client) Update(updateProjectOptions *platform.UpdateProjectOptions) error {
-	return nuclio.ErrNotImplemented
+	c.logger.DebugWith("Sending update project request to leader",
+		"name", updateProjectOptions.ProjectConfig.Meta.Name,
+		"namespace", updateProjectOptions.ProjectConfig.Meta.Namespace)
+
+	// generate request body
+	body, err := c.generateProjectRequestBody(&updateProjectOptions.ProjectConfig)
+	if err != nil {
+		return errors.Wrap(err, "Failed to generate project request body")
+	}
+
+	// send the request
+	headers := c.generateCommonRequestHeaders()
+	responseBody, _, err := common.SendHTTPRequest(http.MethodPut,
+		fmt.Sprintf("%s/%s/%s",
+			c.platformConfiguration.ProjectsLeader.APIAddress,
+			"projects/__name__",
+			updateProjectOptions.ProjectConfig.Meta.Name),
+		body,
+		headers,
+		[]*http.Cookie{updateProjectOptions.SessionCookie},
+		http.StatusOK,
+		true)
+	if err != nil {
+		return errors.Wrap(err, "Failed to send request to leader")
+	}
+
+	c.logger.DebugWith("Successfully sent update project request to leader",
```

Comment:
is the response `Ok`? not `Accepted` with a job to poll on?

Reply:
if it is an immediate update, then you may want to change the log message to `Successfully updated project`. regardless, the "who" updated can be within the log kwargs

Reply:
> is the response Ok? not Accepted with a job to poll on?

Yes. in case of update it's not done inside a job
> if it is an immediate update, then you may want to change the log message to Successfully updated project. regardless, the "who" updated can be within the log kwargs

it's an immediate update but it's not the project creation on nuclio, which will be called during the update on zebo.
I think we should keep it this way, to separate the creation success on zebo and on nuclio internally